test/lib/find-scaffolds.js

```diff
@@ -36,45 +35,67 @@ const dependencies = {
     name: 'incorrectly-tagged',
     realPath: 'incorrectly-tagged realpath',
     keywords: ['cloverfield', 'something else']
+  },
+  packageWithoutName: {
+    // should skip this because it does not have a name
+    realPath: 'package-without-name realpath',
+    keywords: ['cloverfield', 'scaffold']
   }
 };
 
 
-const before = () => {
-  // Stubs
-  const load = spy();
-
-  stub(npm, 'load', load);
-  npm.commands = {
-    ls: spy()
+const mockNpm = () => {
+  const npm = {
+    load: spy(),
+    commands: {
+      ls: spy()
+    }
   };
 
-  return {load};
+  return {npm};
 };
 
 
-const after = () => {
-  npm.load.restore();
-};
+test('Find Scaffolds API', assert => {
+  const {npm} = mockNpm();
+
+  assert.ok(findGlobal(npm) instanceof Function, 'should be function');
+  assert.ok(findGlobal(npm)() instanceof Promise, 'should return Promise');
```

Comment:
I just ran this test and forced it to fail to give you an example, but I was mistaken.

I favor `.equal()` assertions over `.ok()` because when you use `.equal()` you're explicitly stating in your test the expected value and the the actual value gets printed in a failure report.

The `.ok()` assertion is less precise because it will accept any truthy value. I extended that logic and mistakenly assumed that the `.ok()` value would not show you the actual vs expected output when it fails. It turns out that it does. No worries. =)